qiskit_experiments/library/randomized_benchmarking/rb_analysis.py

```diff
@@ -92,45 +92,51 @@ def _default_options(cls):
 
         return default_options
 
-    def _setup_fitting(self, **options) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
+    def _setup_fitting(self, **extra_options) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
         """Fitter options."""
         user_p0 = self._get_option("p0")
         user_bounds = self._get_option("bounds")
 
         curve_data = self._data()
-        initial_guess = self._initial_guess(curve_data.x, curve_data.y, self._num_qubits)
-        fit_option = {
-            "p0": {
-                "a": user_p0["a"] or initial_guess["a"],
-                "alpha": user_p0["alpha"] or initial_guess["alpha"],
-                "b": user_p0["b"] or initial_guess["b"],
-            },
+
+        initial_guess = self._initial_guess(curve_data.x, curve_data.y, self._num_qubits, user_p0)
+        fit_options = {
+            "p0": initial_guess,
             "bounds": {
                 "a": user_bounds["a"] or (0.0, 1.0),
                 "alpha": user_bounds["alpha"] or (0.0, 1.0),
                 "b": user_bounds["b"] or (0.0, 1.0),
             },
         }
-        fit_option.update(options)
+        # p0 and bounds are defined in the default options, therefore updating
+        # with the extra options only adds options and doesn't override p0 or bounds
+        fit_options.update(extra_options)
 
-        return fit_option
+        return fit_options
 
     @staticmethod
     def _initial_guess(
-        x_values: np.ndarray, y_values: np.ndarray, num_qubits: int
+        x_values: np.ndarray, y_values: np.ndarray, num_qubits: int, user_p0: Dict = None
```

Comment:
Why this is needed to be passed? To calculate guess of `a` based on user provided `alpha`?

Reply:
Sorry, It's written in the PR. However you also need to update IRB analysis.

Reply:
IRB is not updated because there is not information from the user as to what to pass there. The user doesn't provide separate guesses for each of the two runs.

Reply:
Thanks, fixed. Can you please check? @gadial Can you please also check.